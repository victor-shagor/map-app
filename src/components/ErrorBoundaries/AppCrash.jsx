import crashImg from "assets/appcrash.svg";

const AppCrash = ({ error }) => {
  return (
    <div className="app-crash" data-testid="app-crash">
      <div className="app-crash-content">
        <img src={crashImg} width="350" alt="app-crash" />
        <div className="app-crash-details">
          <h2>App Crashed</h2>
          <p>Something went horribly wrong.</p>
          <div>
            <p>our engineers are working on this</p>
          </div>
        </div>
      </div>
    </div>
  );
};

export default AppCrash;
