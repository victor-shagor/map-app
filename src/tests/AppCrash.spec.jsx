import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import AppCrash from "components/ErrorBoundaries/AppCrash";

describe("App crash Component", () => {
  it("render App crash component", () => {
    render(<AppCrash />);
    expect(screen.getByTestId("app-crash")).toBeInTheDocument();
  });

  it("renders correctly", () => {
    const tree = renderer.create(<AppCrash />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
