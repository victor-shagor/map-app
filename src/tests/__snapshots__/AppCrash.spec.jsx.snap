// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App crash Component renders correctly 1`] = `
<div
  className="app-crash"
  data-testid="app-crash"
>
  <div
    className="app-crash-content"
  >
    <img
      alt="app-crash"
      src="appcrash.svg"
      width="350"
    />
    <div
      className="app-crash-details"
    >
      <h2>
        App Crashed
      </h2>
      <p>
        Something went horribly wrong.
      </p>
      <div>
        <p>
          our engineers are working on this
        </p>
      </div>
    </div>
  </div>
</div>
`;
